### ISSUE TITLE
Update Julia version

### ISSUE THREAD

**Issue by szabo137** [QEDjl-project/QEDbase.jl#16]
The compatible Julia version is pinned to `1.6`, but should be updated to at least `1.9`. 

In this regard, the unit-test CI could be updated to, e.g., cover the Julia versions `1.7, 1.8, 1.9`

**Comment by szabo137** [comment]
⚠️ Edit

After some investigation, we found the following practices in more extensive Julia packages (e.g. Plots.jl, ModelingToolkit.jl, Dataframes.jl, Jump.jl):

* in the `[compat]` section of the `Project.toml` the Julia version is pinned to `julia==1.6`, which is the current LTS
* in the ci workflows the unit and integration tests run against `julia=1.6` (LTS), `julia==1` (which automatically inserts the current stable version), and `nightly` (the latest version)

Therefore, I suggest to follow these practices and pinn the version to `1.6` and test against `1.6`, `1`, and `nightly`, where `nightly` should be allowed to fail`.

**Comment by SimeonEhrig** [comment]
I implemented running the unit tests with Julia 1.6 until 1.9 and nightly (can fail) for QEDbase: https://github.com/QEDjl-project/QEDbase.jl/pull/29
This works well. Julia 1.6 until 1.9 is working. The nightly fails. Afterwards I extended the integration tests to run Julia 1.6 until 1.9 and nightly. This failed, because QEDProcesses requires Julia 1.9 in the `[compat]` section in the `Project.toml`: https://gitlab.com/hzdr/qedjl-project/QEDbase-jl/-/jobs/5262999313

Therefore I started to implement unit tests with different versions in all sub packages. Now QEDprocess fails because of a real bug: https://gitlab.com/hzdr/qedjl-project/QEDprocesses-jl/-/jobs/5267650216

How do we want to continue? I suggest support the same Julia minimum version for all packages. Should we use version 1.6 or a newer version?
